contributed_definitions/nyaml/NXsource.yaml

```diff
@@ -0,0 +1,521 @@
+  emittance_y(NX_FLOAT):
+    unit: NX_EMITTANCE
+    doc: |
+      Source emittance (nm-rad) in Y (horizontal) direction.
+  sigma_x(NX_FLOAT):
+    unit: NX_LENGTH
+    doc: |
+      particle beam size in x
+  sigma_y(NX_FLOAT):
+    unit: NX_LENGTH
+    doc: |
+      particle beam size in y
+  flux(NX_FLOAT):
+    unit: NX_FLUX
+    doc: |
+      Source intensity/area (example: s-1 cm-2)
+  energy(NX_FLOAT):
+    unit: NX_ENERGY
+    doc: |
+      Source energy.
+      For storage rings, this would be the particle beam energy.
+      For X-ray tubes, this would be the excitation voltage.
+  current(NX_FLOAT):
+    unit: NX_CURRENT
+    doc: |
+      Accelerator, X-ray tube, or storage ring current
+  voltage(NX_FLOAT):
+    unit: NX_VOLTAGE
+    doc: |
+      Accelerator voltage
+  frequency(NX_FLOAT):
+    unit: NX_FREQUENCY
+    doc: |
+      Frequency of pulsed source
+  period(NX_FLOAT):
+    unit: NX_PERIOD
+    doc: |
+      Period of pulsed source
+  target_material:
+    doc: |
+      Pulsed source target material
+    enumeration: [Ta, W, depleted_U, enriched_U, Hg, Pb, C]
+  notes(NXnote):
+    doc: |
+      any source/facility related messages/events that
+      occurred during the experiment
+  bunch_pattern(NXdata):
+    doc: |
+      For storage rings, description of the bunch pattern.
+      This is useful to describe irregular bunch patterns.
+    title:
+      doc: |
+        name of the bunch pattern
+  number_of_bunches(NX_INT):
+    doc: |
+      For storage rings, the number of bunches in use.
+  bunch_length(NX_FLOAT):
+    unit: NX_TIME
+    doc: |
+      For storage rings, temporal length of the bunch
+  bunch_distance(NX_FLOAT):
+    unit: NX_TIME
+    doc: |
+      For storage rings, time between bunches
+  pulse_width(NX_FLOAT):
+    unit: NX_TIME
+    doc: |
+      temporal width of source pulse
+  
+  # pulsed sources or storage rings could use this
+  pulse_shape(NXdata):
+    doc: |
+      source pulse shape
+  
+  # pulsed sources or storage rings could use this
+  mode:
+    doc: |
+      source operating mode
+    enumeration: 
+      Single Bunch: 
+        doc: |
+          for storage rings
+      Multi Bunch: 
+        doc: |
+          for storage rings
+  
+  # other sources could add to this
+  
+  # other sources could add to this
+  top_up(NX_BOOLEAN):
+    doc: |
+      Is the synchrotron operating in top_up mode?
+  last_fill(NX_NUMBER):
+    unit: NX_CURRENT
+    doc: |
+      For storage rings, the current at the end of the most recent injection.
+    \@time:
+      type: NX_DATE_TIME
+      doc: |
+        date and time of the most recent injection.
+  photon_energy(NX_FLOAT):
+    unit: NX_ENERGY
+    doc: |
+      The center photon energy of the source, before it is
+      monochromatized or converted
+  center_wavelength(NX_FLOAT):
+    unit: NX_WAVELENGTH
+    doc: |
+      The center wavelength of the source, before it is
+      monochromatized or converted
```

Comment:
@Tommaso-Pincelli  Can't we just use energy for this, and extend it's docstring?

Reply:
Note that whatever is agreed here, the changes should be made in `base_classes/nyaml/NXsource.yaml` since `NXsource` already exists in the NIAC repo.

Reply:
Since we (@lukaspie, @rettigl and I) discussed to mainly rely on NXbeam to collect relevant metadata about the beam characteristics I would argue that we should not extend `NXsource` with beam related fields here

Reply:
I don't understand why `contributed_definitions/nyaml/NXsource.yaml` is back here. I had deleted it manually because `NXsource `already exists in the NIAC repo (and thus, changes must be done in `base_classes/nyaml/NXsource.yaml`). There exists no `contributed_definitions/NXsource.nxdl.xml` so I cannot be because of automatic yaml generation.

Reply:
Or is this because it still has this open conversation?

Reply:
The comment is on an old commit. So it refers to the file in the _history_ of this repository. In the current state of the repo neither `contributed_definitions/NXsource.nxdl.xml` nor `contributed_definitions/nyaml/NXsource.yaml` are present.

Reply:
That's what this small outdated says:

<img width="86" alt="Bildschirmfoto 2024-01-15 um 16 55 52" src="https://github.com/FAIRmat-NFDI/nexus_definitions/assets/1014422/06b7c636-f8a7-4897-b4d6-45056b5745a2">

it means that the current line does not exist anymore. this might be due to that the line has changed, removed or the file is not present anymore at all.

Reply:
Ok thanks, it just looked weird in the files tab. GitHub confuses me sometimes :D 

Nevertheless, even in the `NXsource` base class, we have added a lot of fields compared to the [NIAC version](https://github.com/nexusformat/definitions/blob/main/base_classes/NXsource.nxdl.xml). They have been added [here](https://github.com/FAIRmat-NFDI/nexus_definitions/commit/d94a5cef37f6b59414218aa609f005fa3d3a4578). The question is whether we actually need these fields. I would argue we don't if we want to describe everything related to the beam characteristics in `NXbeam`.

Reply:
Ok, I see. These changes are already in fairmat. So I would keep them out here and open a separate PR if we want to remove them again.

Reply:
Sounds good. Apparently, at least some of these new fields are also used in [APM](https://fairmat-nfdi.github.io/nexus_definitions/classes/contributed_definitions/NXapm.html), so we shouldn't just blindly remove them. I have created an [issue](https://github.com/FAIRmat-NFDI/nexus_definitions/issues/145) and will resolve here.

Reply:
I agree that this is a separate discussion.